src/sandstorm/grain.capnp

```diff
@@ -160,27 +160,11 @@ interface SandstormApi(AppObjectId) {
   #   failure.
 }
 
-struct DenormalizedGrainMetadata @0xbdd9bea5585df6c5 {
-  # The metadata that we need to present contextual information for shared grains (in particular,
-  # information about the app providing that grain, like icon and title).
+interface StaticAsset @0xfabb5e621fa9a23f {
+  # A file served by the Sandstorm frontend, suitable for embedding e.g. in <img> elements
+  # inside of a grain iframe.
 
-  appTitle @0 :Util.LocalizedText;
-  # A copy of the app name for the corresponding UIView for presentation in the grain list.
-
-  union {
-    icon :group {
-      format @1 :Text;
-      # Icon asset format, if present.  One of "png" or "svg"
-
-      assetId @2 :Text;
-      # The asset ID associated with the grain-size icon for this token
-
-      assetId2xDpi @3 :Text;
-      # If present, the asset ID for the equivalent asset as assetId at twice-resolution
-    }
-    appId @4 :Text;
-    # App ID, needed to generate a favicon if no icon is provided.
-  }
+  getUrl @0 () -> (url :Text);
```

Comment:
This is XSS-prone: If the StaticAsset capability is implemented by another app or other malicious source, it can return a `javascript:` URL or similar, which is likely to be evaluated in the context of the calling app. You can say it is the caller's responsibility to filter, but most won't.

We can prevent this problem by changing the return to something like:

    (https: Bool, hostPath :Text)

The URL is then formed as:

    (https ? "https://" : "http://") + hostPath